src/editor/underlying.rs

```diff
@@ -232,17 +104,63 @@ impl ObjectImpl for EditorWindow {
         warn!("Image status {:?}", image.status());
 
         let overlay = gtk::Overlay::new();
+        obj.add(&overlay);
         let drawing_area = gtk::DrawingArea::builder()
             .can_focus(true)
             .events(EventMask::ALL_EVENTS_MASK)
             .build();
-        let button = gtk::Button::new();
-        let toolbar = gtk::Box::new(gtk::Orientation::Horizontal, 12);
-        obj.add(&overlay);
-        overlay.add(&drawing_area);
-        button.set_label("long text");
 
-        toolbar.add(&button);
+        let toolbar = gtk::Box::new(gtk::Orientation::Horizontal, 0);
+
+        let rectangle_crop = gtk::Button::builder().label("RectC").build();
```

Comment:
This entire section has a lot of repeated code. How about a method to which you pass the toolbar, a string for the label, and the appropriate `Tool`? (Am not adding the `image` to the parameter list since it is inside our struct already)

Reply:
Yeah I think that'd be a better solution

Reply:
:+1: